src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2789,6 +2789,37 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
             return CreatePointerTypeSymbol(elementType.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>("elementType"));
         }
 
+        protected override INamedTypeSymbol CommonCreateTupleTypeSymbol(ITypeSymbol[] elementTypes, string[] elementNames)
+        {
+            if (elementTypes.Length <= 1)
+            {
+                throw new ArgumentException(CodeAnalysisResources.TuplesNeedAtLeastTwoElements, nameof(elementNames));
+            }
+
+            if (elementNames != null && elementNames.Length != elementTypes.Length)
+            {
+                throw new ArgumentException(CodeAnalysisResources.TupleNamesAllOrNone, nameof(elementNames));
+            }
+
+            var typesBuilder = ArrayBuilder<TypeSymbol>.GetInstance(elementTypes.Length);
+            for (int i = 0; i < elementTypes.Length; i++)
+            {
+                if (elementTypes[i] == null)
+                {
+                    throw new ArgumentNullException($"{nameof(elementTypes)}[{i}]");
+                }
+
+                typesBuilder.Add(elementTypes[i].EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>(nameof(elementTypes)));
+            }
+
+            if (elementNames == null)
+            {
+                return TupleTypeSymbol.Create(typesBuilder.ToImmutableAndFree(), default(ImmutableArray<string>), this);
+            }
+
+            return TupleTypeSymbol.Create(typesBuilder.ToImmutableAndFree(), elementNames.ToImmutableArrayOrEmpty(), this);
```

Comment:
After changing `elementNames` to `ImmutableArray<string>`, I can pass it directly into `TupleTypeSymbol` whether it have values or it is default.